### DISCUSSION TITLE
What is the use case for not associating a VNet to a block?

### DISCUSSION THREAD

**Discussion by whizzer929** [Azure/ipam#77] [General]
Hi Dev Team,
I'm curious what the use case would be for _NOT_ associating a VNet to a Block?  We are accomplishing this by using reservations and the tag on the VNet, which works great btw.  But if someone were to add a VNet manually, we have to associate it manually.  
Could we add a feature request to automatically associate all VNets if they fall within the Blocks IP Space?  Again, not sure why you wouldn't do this, but I'm sure there is a reason.

Thanks!

**Comment by DCMattyG** [comment]
Hey there @whizzer929, anything can be done if it makes sense haha!

I believe the quick answer right now is that in the current design, there cannot be overlapping vNET's within a Block/Space.....so in your example above, what would happen if 7 people create a vNET with the Address Space of 10.0.0.0/16? How would overlaps get handled?

The design is as such with the intent of having "managed" vNET's that live within a given Block/Space, and "unmanaged" vNET's that would ideally be migrated into a Block/Space, or remediated if they are overlapping.

Given those details, let me know your thoughts on how everything should work ideally, and let's talk about how w get there together!

**Reply by whizzer929** [reply]
Hi Matt, yeah that makes perfect sense.  We haven't run into that too much as we filter noisy and sandbox subscriptions out from IPAM.  In our case, only VNets that are peered to the VWan hub are in scope for our IPAM.  I guess as a feature request, maybe have a query built into the automatic association, I.E. Peer name = "VWan Hub", or something like that.    This is no doubt complicated and this isn't an urgent need.  

I would like to make a request though that on the Virtual Network Association page, the Subscription be the DisplayName and not the ID, or both.  Thanks again!

![2023-02-07_13-46-10](https://user-images.githubusercontent.com/65122209/217374049-d9f21162-37b5-4fcb-861a-a4ab9882fe5c.jpg)

**Reply by DCMattyG** [reply]
Thanks @whizzer929!

One the first item, let's definitely work together on this!

For the second item, I'm on it!
